File: back/src/main/java/com/back/domain/member/service/AuthTokenService.java
package com.back.domain.member.service;

import com.back.domain.member.entity.Member;
import com.back.standard.util.Ut;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

import java.util.Map;

@Service
public class AuthTokenService {
    @Value("${custom.accessToken.expirationSeconds}")
    private int expireSeconds;

    @Value("${custom.jwt.secretKey}")
    private String secret;

    String genAccessToken(Member member) {
        long id = member.getId();
        String username = member.getUsername();
        String name = member.getName();
        String role = member.getRole().name();

        return Ut.jwt.toString(
                secret,
                expireSeconds,
                Map.of("id", id, "username", username, "name", name, "role", role)
        );

    }

    Map<String, Object> payload(String accessToken) {
        Map<String, Object> parsedPayload = Ut.jwt.payload(secret, accessToken);

        if (parsedPayload == null) return null;

        int id = (int) parsedPayload.get("id");
        String username = (String) parsedPayload.get("username");
        String name = (String) parsedPayload.get("name");
        String role = (String) parsedPayload.get("role");

        return Map.of("id", id, "username", username, "name", name, "role", role);
    }
}

File: back/src/main/kotlin/com/back/domain/member/service/AuthTokenService.kt
package com.back.domain.member.service

import com.back.domain.member.entity.Member
import com.back.standard.util.Ut
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service

@Service
class AuthTokenService {
    @Value("\${custom.accessToken.expirationSeconds}")
    private var expireSeconds: Int = 0

    @Value("\${custom.jwt.secretKey}")
    private lateinit var secret: String

    fun genAccessToken(member: Member): String {
        val id = member.id
        val username = member.username
        val name = member.name
        val role = member.role.name

        return Ut.jwt.toString(
            secret,
            expireSeconds,
            mapOf("id" to id, "username" to username, "name" to name, "role" to role)
        )
    }

    fun payload(accessToken: String): Map<String, Any>? {
        val parsedPayload = Ut.jwt.payload(secret, accessToken) ?: return null

        val id = parsedPayload["id"] as Int
        val username = parsedPayload["username"] as String
        val name = parsedPayload["name"] as String
        val role = parsedPayload["role"] as String

        return mapOf("id" to id, "username" to username, "name" to name, "role" to role)
    }
}